jaxdecomp/_src/fft.py

```diff
@@ -75,46 +81,38 @@ def sfft_abstract_eval(x, fft_type, pdims, global_shape, adjoint):
     axis = 2
 
   output_shape = None
+  match fft_type:
+    case xla_client.FftType.FFT:
+      # FFT is X to Y to Z so Z-Pencil is returned
+      # Except if we are doing a YZ slab in which case we return a Y-Pencil
+      transpose_shape = (1, 2, 0) if not yz_slab else (2, 0, 1)
+      transposed_pdims = pdims if not yz_slab else pdims[::-1]
+    case xla_client.FftType.IFFT:
+      # IFFT is Z to X to Y so X-Pencil is returned
+      # In YZ slab case we only need one transposition back to get the X-Pencil
+      transpose_shape = (2, 0, 1) if not yz_slab else (1, 2, 0)
+      transposed_pdims = pdims
```

Comment:
Why is the forward FFT the only one for which you reverse the pdims with `yz_slab` ?

Reply:
In case we have pencils we transpose twice (From X to Y from  Y to Z)
In case of XY slab we do a 2D FFT on XY and then transpose to Y then from Y to Z (can't transpose from X to Z because of the cyclic shift that cudecomp does) so again two transposes
In case of XZ slab (so X , Y / 4 , Z) for example, we do 1D FFT on X then transpose X to Y then 2D FFT on YZ ... only one transpose is needed 
In the other PR I explain that every one transpose the pdims are switch, every two transposes they are the same